argocd-server crashes v2.0.4

If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

argocd-server crashes and there doesn't seem to be a specific action that any user did to cause it.

**To Reproduce**

I'm not entirely sure because it seemed to happen out of the blue. I'm guessing somehow ArgoCD server got in a bad state. I had just run a manual sync for our Application "cortex" so that is what you see finishing up around 12:55 in the logs. Then the crash happened around 

**Expected behavior**

argocd-server does not crash

**Screenshots**

n/a

**Version**

```shell
🚀 k exec -it argocd-server-6589549db5-jt5kw -- argocd version
argocd: v2.0.4+0842d44
  BuildDate: 2021-06-23T01:27:53Z
  GitCommit: 0842d448107eb1397b251e63ec4d4bc1b4efdd6e
  GitTreeState: clean
  GoVersion: go1.16
  Compiler: gc
  Platform: linux/amd64
FATA[0000] Argo CD server address unspecified
command terminated with exit code 1
```

I'm not sure why I cannot connect to the server when I'm running from within the pod, but the server version is that same version seeing as thought I ran the command using the binary from within the argocd-server container 😄 

**Logs**

I have all of the logs if they are helpful, but figured not. Let me know if you want the full log.

```
> k logs argocd-server-6589549db5-jt5kw --previous
...
time="2021-07-16T12:55:45Z" level=info msg="Ignore status for CustomResourceDefinitions"
time="2021-07-16T12:55:45Z" level=info msg="finished unary call with code OK" grpc.code=OK grpc.method=Get grpc.service=cluster.SettingsService grpc.start_time="2021-07-16T12:55:45Z" grpc.time_ms=0.205 span.kind=server system=grpc
time="2021-07-16T12:55:58Z" level=info msg="Alloc=21988 TotalAlloc=34795437 Sys=78097 NumGC=2624 Goroutines=182"
time="2021-07-16T12:59:18Z" level=info msg="finished streaming call with code OK" grpc.code=OK grpc.method=Watch grpc.service=application.ApplicationService grpc.start_time="2021-07-16T12:54:48Z" grpc.time_ms=270002.3 span.kind=server system=grpc
time="2021-07-16T12:59:40Z" level=info msg="received streaming call /application.ApplicationService/Watch" grpc.method=Watch grpc.request.claims=null grpc.request.content="name:\"cortex\" resourceVersion:\"\" selector:\"\" repo:\"\" " grpc.service=application.ApplicationService grpc.start_time="2021-07-16T12:59:40Z" span.kind=server system=grpc
time="2021-07-16T12:59:43Z" level=info msg="finished streaming call with code OK" grpc.code=OK grpc.method=WatchResourceTree grpc.service=application.ApplicationService grpc.start_time="2021-07-16T12:53:12Z" grpc.time_ms=391345 span.kind=server system=grpc
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x56df82]

goroutine 1310682 [running]:
bufio.(*Writer).Flush(0xc0017d0100, 0x1, 0x7fc88a417438)
        /usr/local/go/src/bufio/bufio.go:607 +0x62
net/http.(*response).Flush(0xc000de8c40)
        /usr/local/go/src/net/http/server.go:1681 +0x35
github.com/argoproj/pkg/grpc/http.writeKeepalive(0x2d6ce40, 0xc000de8c40, 0xc0015bc788)
        /go/pkg/mod/github.com/argoproj/pkg@v0.9.1-0.20210512035321-be5ba22dca5b/grpc/http/forwarders.go:151 +0x182
github.com/argoproj/pkg/grpc/http.keepalive(0x2d74518, 0xc001ce14d0, 0x2d6ce40, 0xc000de8c40, 0xc0015bc788)
        /go/pkg/mod/github.com/argoproj/pkg@v0.9.1-0.20210512035321-be5ba22dca5b/grpc/http/forwarders.go:166 +0x9f
created by github.com/argoproj/pkg/grpc/http.withKeepalive
        /go/pkg/mod/github.com/argoproj/pkg@v0.9.1-0.20210512035321-be5ba22dca5b/grpc/http/forwarders.go:174 +0x9a
```

```
> k describe po argocd-server-6589549db5-jt5kw
...
Containers:
  argocd-server:
    Container ID:  docker://4642a0ddc7da97f9d0d5a5f560d120ca261f4ccb1b9a18d77c363c540c64fd7e
    Image:         xxx.dkr.ecr.us-east-1.amazonaws.com/infra-mirror/quayio_argoproj_argocd:v2.0.4
    Image ID:      docker-pullable://xxx.dkr.ecr.us-east-1.amazonaws.com/infra-mirror/quayio_argoproj_argocd@sha256:976dfbfadb817ba59f4f641597a13df7b967cd5a1059c966fa843869c9463348
    Ports:         8080/TCP, 8083/TCP
    Host Ports:    0/TCP, 0/TCP
    Command:
      argocd-server
      --staticassets
      /shared/app
      --redis
      argocd-redis-ha-haproxy:6379
      --insecure
    State:          Running
      Started:      Fri, 16 Jul 2021 08:59:44 -0400
    Last State:     Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Tue, 13 Jul 2021 08:55:58 -0400
      Finished:     Fri, 16 Jul 2021 08:59:43 -0400
    Ready:          True
    Restart Count:  1
    Liveness:       http-get http://:8080/healthz%3Ffull=true delay=3s timeout=1s period=30s #success=1 #failure=3
    Readiness:      http-get http://:8080/healthz delay=3s timeout=1s period=30s #success=1 #failure=3
    Environment:
      ARGOCD_GPG_ENABLED:           false
      ARGOCD_API_SERVER_REPLICAS:   2
      AWS_ROLE_ARN:                 arn:aws:iam::xxx:role/xxx
      AWS_WEB_IDENTITY_TOKEN_FILE:  /var/run/secrets/eks.amazonaws.com/serviceaccount/token
    Mounts:
      /app/config/server/tls from argocd-repo-server-tls (rw)
      /app/config/ssh from ssh-known-hosts (rw)
      /app/config/tls from tls-certs (rw)
      /var/run/secrets/eks.amazonaws.com/serviceaccount from aws-iam-token (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from argocd-server-token-6sjgp (ro)
...
```

--- a/go.mod
+++ b/go.mod
@@ -8,7 +8,7 @@ require (
 	github.com/alicebob/miniredis v2.5.0+incompatible
 	github.com/alicebob/miniredis/v2 v2.14.2
 	github.com/argoproj/gitops-engine v0.3.1-0.20210709004906-a4c77d5c70fb
-	github.com/argoproj/pkg v0.9.1-0.20210512035321-be5ba22dca5b
+	github.com/argoproj/pkg v0.9.1
 	github.com/bombsimon/logrusr v1.0.0
 	github.com/bradleyfalzon/ghinstallation v1.1.1
 	github.com/casbin/casbin v1.9.1

